digdag-standards/src/main/java/io/digdag/standards/command/ecs/EcsClientConfig.java

```diff
@@ -2,36 +2,71 @@
 
 import com.google.common.base.Optional;
 import io.digdag.client.config.Config;
-import io.digdag.client.config.ConfigException;
 import io.digdag.core.storage.StorageManager;
 
-import java.util.Arrays;
 import java.util.List;
 
 public class EcsClientConfig
 {
     private static final String SYSTEM_CONFIG_PREFIX = "agent.command_executor.ecs.";
+    public static final String TASK_CONFIG_ECS_KEY = "agent.command_executor.ecs";
+    private static final int DEFAULT_MAX_TRIES = 3;
 
     public static EcsClientConfig of(final Optional<String> clusterName, final Config systemConfig, final Config taskConfig)
     {
-        return EcsClientConfig.createFromSystemConfig(clusterName, systemConfig);
-        /**
-        if (config.has("ecs")) {
+        if (taskConfig.has(TASK_CONFIG_ECS_KEY)) {
             // from task config
-            return createFromTaskConfig(clusterName, config); // TODO
+            return createFromTaskConfig(clusterName, taskConfig);
         }
         else {
             // from system config
-            return EcsClientConfig.createFromSystemConfig(clusterName, systemConfig);
+            return createFromSystemConfig(clusterName, systemConfig);
         }
-         */
+    }
```

Comment:
`EcsClientConfig.of` is an entry point of this PR's change.

Currently, `EcsClientConfig.of` always creates `EcsClientConfig` object from `systemConfig`.
In this PR, if `taskConfig` has key of `agent.command_executor.ecs`, `EcsClientConfig.of` will be created from `taskConfig`.

Reply:
Refactored here 4faf4b0